Frontend/HsParser.hs

```diff
@@ -281,12 +281,33 @@ pTransPat (PsdAppPat p1 p2) = do
   p2' <- pTransPat p2
   return $ HsConPat dc (xs ++ [p2']) -- NOTE: this is O(n^2) because of the (++), could be O(n)
 
+-- | Wrap the contents of a monad into a singleton array
+one :: Monad f => f a -> f [a]
+one x = x >>= (\x' -> return [x'])
```

Comment:
Alternatively:
```suggestion
one x = pure <$> x
```
which means that you can inline this function completely :wink:

Reply:
Damnit I searched so hard to make it smallest but I knew there was going to be something shorter that I had missed haha 😅